src/lib/syncable_ledger/test/test.ml

```diff
@@ -35,6 +37,64 @@ module type Input_intf = sig
        and type answer := (Root_hash.t, Ledger.account) Syncable_ledger.Answer.t
 end
 
+module Make_context (Subtree_depth : sig
+  val sync_ledger_max_subtree_depth : int
+
+  val sync_ledger_default_subtree_depth : int
+end) : Syncable_ledger.CONTEXT = struct
+  let logger = Logger.null ()
+
+  let compile_config =
+    { Mina_compile_config.For_unit_tests.t with
+      sync_ledger_max_subtree_depth =
+        Subtree_depth.sync_ledger_max_subtree_depth
+    ; sync_ledger_default_subtree_depth =
+        Subtree_depth.sync_ledger_default_subtree_depth
+    }
+end
+
+module Context_subtree_depth32 = Make_context (struct
+  let sync_ledger_max_subtree_depth = 3
+
+  let sync_ledger_default_subtree_depth = 2
+end)
+
+module Context_subtree_depth81 = Make_context (struct
+  let sync_ledger_max_subtree_depth = 8
+
+  let sync_ledger_default_subtree_depth = 1
+end)
+
+module Context_subtree_depth82 = Make_context (struct
+  let sync_ledger_max_subtree_depth = 8
+
+  let sync_ledger_default_subtree_depth = 2
+end)
+
+module Context_subtree_depth86 = Make_context (struct
+  let sync_ledger_max_subtree_depth = 8
+
+  let sync_ledger_default_subtree_depth = 6
+end)
+
+module Context_subtree_depth88 = Make_context (struct
+  let sync_ledger_max_subtree_depth = 8
+
+  let sync_ledger_default_subtree_depth = 8
+end)
+
+module Context_subtree_depth68 = Make_context (struct
+  let sync_ledger_max_subtree_depth = 6
+
+  let sync_ledger_default_subtree_depth = 8
+end)
+
+module Context_subtree_depth80 = Make_context (struct
+  let sync_ledger_max_subtree_depth = 8
+
+  let sync_ledger_default_subtree_depth = 0
```

Comment:
This test is a strange one: shouldn't we disallow `default_subtree_depth = 0`? If we request depth `0`, I would understand that as meaning that we expect to get back `[|hash|]` when we request the depth-0 children of `hash`.

Reply:
Yes, it is disallowed. It was an easy way to test the case. I could add a top level assertion for this in node_config and runtime config so that nodes aren't configured with 0 sub tree depth